--- origsrc/Python-2.5.4/configure.in	2008-12-13 08:13:52.000000000 -0600
+++ src/Python-2.5.4/configure.in	2009-03-16 01:15:57.968750000 -0500
@@ -1543,7 +1543,7 @@ then
 		fi;;
 	SCO_SV*) LDSHARED='$(CC) -Wl,-G,-Bexport';;
 	Monterey*) LDSHARED="cc -G -dy -Bdynamic -Bexport -L/usr/lib/ia64l64";;
-	CYGWIN*) LDSHARED="gcc -shared -Wl,--enable-auto-image-base";;
+	CYGWIN*) LDSHARED='$(CC) -shared -Wl,--enable-auto-image-base';;
 	atheos*) LDSHARED="gcc -shared";;
 	*)	LDSHARED="ld";;
 	esac
