--- origsrc/Python-2.5.2/Modules/ossaudiodev.c	2006-10-04 05:23:57.000000000 -0500
+++ src/Python-2.5.2/Modules/ossaudiodev.c	2008-07-01 21:08:33.531250000 -0500
@@ -841,7 +841,7 @@
 }
 
 static PyTypeObject OSSAudioType = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,                          /*ob_size*/
     "ossaudiodev.oss_audio_device", /*tp_name*/
     sizeof(oss_audio_t),        /*tp_size*/
@@ -856,7 +856,7 @@
 };
 
 static PyTypeObject OSSMixerType = {
-    PyObject_HEAD_INIT(&PyType_Type)
+    PyObject_HEAD_INIT(NULL)
     0,                              /*ob_size*/
     "ossaudiodev.oss_mixer_device", /*tp_name*/
     sizeof(oss_mixer_t),            /*tp_size*/
--- origsrc/Python-2.5.2/setup.py	2008-02-04 17:41:02.000000000 -0600
+++ src/Python-2.5.2/setup.py	2008-07-02 02:11:28.671875000 -0500
@@ -1062,7 +1062,7 @@
             exts.append( Extension('linuxaudiodev', ['linuxaudiodev.c']) )
 
         if platform in ('linux2', 'freebsd4', 'freebsd5', 'freebsd6',
-                        'freebsd7'):
+                        'freebsd7', 'cygwin'):
             exts.append( Extension('ossaudiodev', ['ossaudiodev.c']) )
 
         if platform == 'sunos5':
