--- origsrc/Python-2.6.5/Include/py_curses.h	2009-09-06 16:23:05.000000000 -0500
+++ src/Python-2.6.5/Include/py_curses.h	2010-04-14 15:21:23.008971400 -0500
@@ -17,6 +17,13 @@
 #define NCURSES_OPAQUE 0
 #endif /* __APPLE__ */
 
+#ifdef __CYGWIN__
+/* the following define is necessary for Cygwin; without it, the
+   Cygwin-supplied ncurses.h sets NCURSES_OPAQUE to 1, and then Python
+   can't get at the WINDOW flags field. */
+#define NCURSES_INTERNALS
+#endif /* __CYGWIN__ */
+
 #ifdef __FreeBSD__
 /*
 ** On FreeBSD, [n]curses.h and stdlib.h/wchar.h use different guards
--- origsrc/Python-2.6.5/setup.py	2010-04-14 14:33:42.954385600 -0500
+++ src/Python-2.6.5/setup.py	2010-04-14 15:03:50.941796600 -0500
@@ -1776,6 +1776,11 @@ class PyBuildExt(build_ext):
         if not '--with-system-ffi' in sysconfig.get_config_var("CONFIG_ARGS"):
             return
 
+        if sys.platform == 'cygwin':
+            # provided by GCC
+            ext.libraries.append('ffi')
+            self.use_system_libffi = True
+
         if sys.platform == 'darwin':
             # OS X 10.5 comes with libffi.dylib; the include files are
             # in /usr/include/ffi
